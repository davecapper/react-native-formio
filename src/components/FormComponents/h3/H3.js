import React from 'react';
import {Text} from '@rneui/themed';
import PropTypes from 'prop-types';
import styles from './styles';

const H3 = (props) => (
  <Text style={styles.header3}>
    {props.children}
  </Text>
);

H3.propTypes = {
  children: PropTypes.node
};

export default H3;

