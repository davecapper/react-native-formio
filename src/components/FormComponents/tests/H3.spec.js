import React from 'react';
import renderer from 'react-test-renderer';
import H3 from '../h3/H3';
import {Text} from '@rneui/themed';

describe('H3', () => {
  describe(' H3 component', () => {
    it('Renders a basic H3 component', () => {
      const element = renderer.create(<H3><Text>Header</Text></H3>);

      expect(element).toMatchSnapshot();
    });
  });
});
