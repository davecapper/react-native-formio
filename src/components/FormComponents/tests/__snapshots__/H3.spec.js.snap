// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H#  H3 component Renders a basic H3 component 1`] = `
<Text
  accessibilityRole="text"
  style={
    {
      "color": "#242424",
      "fontSize": 22,
      "letterSpacing": 0.35,
      "lineHeight": 28,
      "margin": 10,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#242424",
      }
    }
  >
    Header
  </Text>
</Text>
`;
